Util/src/main/java/io/deephaven/util/mutable/MutableInt.java

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright (c) 2016-2024 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.util.mutable;
+
+/**
+ * Minimal mutable wrapper for an {@code int} value. Loosely based on
+ * {@code org.apache.commons.lang3.mutable.MutableInt}, but without inheriting from {@link Number}, or providing any
+ * overloads that accept {@code Number} or any boxed types.
+ * <p>
+ * Deliberately does not extend {@code Number}, does not implement {@code toString()}/{@code equals}/{@code hashcode()},
+ * or implement {@code Comparable}.
+ */
+
+public class MutableInt {
+    private int value;
+
+    public MutableInt() {
+
+    }
+
+    public MutableInt(final int value) {
+        this.value = value;
+    }
+
+    public int intValue() {
+        return value;
+    }
+
+    public void setValue(int value) {
+        this.value = value;
+    }
+
+    public void add(int addend) {
+        this.value += addend;
+    }
+
+    public int addAndGet(int addend) {
+        value += addend;
+        return value;
+    }
+
+    public int getAndAdd(int addend) {
+        int old = intValue();
```

Comment:
Still have a method call where we don't really need it, but I'm quite sure I trust the JIT to handle this.